.github/workflows/github_pages.yml

```diff
@@ -0,0 +1,64 @@
+name: Deploy to GitHub Pages
+on:
+  workflow_call:
+    inputs:
+      settings:
+        type: string
+        default: "publishconf.py"
+      output-path:
+        type: string
+        default: "output"
+      cache:
+        type: boolean
+      cache-path:
+        type: string
+        default: "cache"
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+      - name: Configure GitHub Pages
+        id: pages
+        uses: actions/configure-pages@v3
+      - name: Install requirements
+        run: pip install "pelican[markdown]"
```

Comment:
Might want to add a workflow option for the `pelican[markdown]` string here so that users can override it to supply their own optional packages or to use a specific version of pelican. Users would do something like this in their `pelican.yml` files:

```yml
with:
  requirements: "pelican[markdown]==4.8.0 typogrify"
```

Reply:
Done